include/ast.hpp

```diff
@@ -220,17 +220,34 @@ struct CompoundStmtNode : public StmtNode {
   std::vector<std::unique_ptr<StmtNode>> stmts;
 };
 
-/// @brief Root of the entire program.
-struct ProgramNode : public AstNode {
+/// @brief An external declaration can be a definition of a function or an
+/// object.
+struct ExternDeclNode : public AstNode {
+  ExternDeclNode(
+      Location loc,
+      std::variant<std::unique_ptr<FuncDefNode>, std::unique_ptr<DeclStmtNode>>
+          extern_decl)
+      : AstNode{loc}, extern_decl{std::move(extern_decl)} {}
+
+  void Accept(NonModifyingVisitor&) const override;
+  void Accept(ModifyingVisitor&) override;
+
+  std::variant<std::unique_ptr<FuncDefNode>, std::unique_ptr<DeclStmtNode>>
+      extern_decl;
+};
+
+/// @brief A translation unit, which the compiler handles individually,
+/// representing a high-level entity in the compilation process.
+struct TransUnitNode : public AstNode {
   /// @note vector of move-only elements are move-only
-  ProgramNode(Location loc,
-              std::vector<std::unique_ptr<FuncDefNode>> func_def_list)
-      : AstNode{loc}, func_def_list{std::move(func_def_list)} {}
+  TransUnitNode(Location loc,
+                std::vector<std::unique_ptr<ExternDeclNode>> trans_unit)
```

Comment:
Does the name `trans_unit` seem more suitable than `extern_decl` from your perspective?